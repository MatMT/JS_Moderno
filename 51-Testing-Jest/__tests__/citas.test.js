import Citas from "../js/classes/Citas";

describe('Probar la clase de Citas', () => {

    const citas = new Citas;
    const id = Date.now();

    test('should Nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Mat',
            telefono: '22224444',
            fecha: '24-11-2023',
            hora: '10:50',
            sintomas: 'Sleepy'
        }

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });

    test('should Actualizar Cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Happy',
            propietario: 'Mat',
            telefono: '22224444',
            fecha: '24-11-2023',
            hora: '10:50',
            sintomas: 'Sleepy'
        }

        citas.editarCita(citaActualizada);

        // Prueba
        expect(citas).toMatchSnapshot();
    })

    test('should Eliminar Cita', () => {
        citas.eliminarCita(id);

        // Prueba
        expect(citas).toMatchSnapshot();
    })
})