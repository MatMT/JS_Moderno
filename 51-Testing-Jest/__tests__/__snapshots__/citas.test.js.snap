// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas should Actualizar Cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "24-11-2023",
      "hora": "10:50",
      "id": 1700888553011,
      "mascota": "Happy",
      "propietario": "Mat",
      "sintomas": "Sleepy",
      "telefono": "22224444",
    },
  ],
}
`;

exports[`Probar la clase de Citas should Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`Probar la clase de Citas should Nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "24-11-2023",
      "hora": "10:50",
      "id": 1700888553011,
      "mascota": "Hook",
      "propietario": "Mat",
      "sintomas": "Sleepy",
      "telefono": "22224444",
    },
  ],
}
`;
